### PR TITLE
JAMES-3439: Email/set create specify normal attachment

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 1]
```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+    Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
+      .fold(_ => None, attachment => Some(attachment))
+  }
+
+  private def loadAttachment(attachmentMetadata: AttachmentMetadata, attachmentContentLoader: AttachmentContentLoader, mailboxSession: MailboxSession): Either[ErrorCreatingAttachmentException, (AttachmentMetadata, Array[Byte])] = {
+    Try(attachmentContentLoader.load(attachmentMetadata, mailboxSession))
+      .toEither
+      .fold(e => {
+        //TODO log error
+        Left(ErrorCreatingAttachmentException(e.getMessage))
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:180]: Why log here if the error is carried over?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:180]: Don't swallow the stacktrace...

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 2]
```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+    val inlineAttachments = attachments.filter(_.isInline)
+    val otherAttachments = attachments.filter(!_.isInline)
+
+    if (inlineAttachments.nonEmpty) {
+      val relatedMultipartBuilder: MultipartBuilder = MultipartBuilder.create(SubType.RELATED_SUBTYPE)
+      relatedMultipartBuilder.addBodyPart(bodypartBuilder.setBody(maybeHtmlBody.getOrElse(""), SubType.HTML_SUBTYPE, StandardCharsets.UTF_8))
+
+      inlineAttachments
+        .flatMap(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+        .foreach(attachmentMetadata => relatedMultipartBuilder.addBodyPart(
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:146]: Use fold?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 3]
```diff
@@ -1470,6 +1471,100 @@ trait EmailSetMethodContract {
...
+      .isEqualTo(
+        s"""[{
+           |  "mailboxIds": {
+           |    "${mailboxId.serialize}": true
+           |  },
+           |  "subject": "World domination",
+           |  "attachments": [
+           |    {
+           |      "partId": "2",
+           |      "blobId": "1_2",
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:1558]: We would need to download that blob to assert that the content is identical...

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 4]
```diff
@@ -61,6 +71,18 @@ case class ClientEmailBodyValue(value: String,
...
+case class Attachment(blobId: BlobId,
+                      `type`: Type,
+                      name: Option[Name],
+                      charset: Option[Charset],
+                      disposition: Option[Disposition],
+                      language: Option[Language],
+                      location: Option[Location],
+                      cid: Option[String]) {
+
+  def isInline: Boolean = disposition.forall(disposition => disposition.value.equals("inline"))
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:83]: disposition.contains("inline") ?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 5]
```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+    from.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setFrom)
+    to.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setTo)
+    cc.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setCc)
+    bcc.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setBcc)
+    sender.flatMap(_.asMime4JMailboxList).map(_.asJava).map(Fields.addressList(FieldName.SENDER, _)).foreach(builder.setField)
+    replyTo.flatMap(_.asMime4JMailboxList).map(_.asJava).foreach(builder.setReplyTo)
+    sentAt.map(_.asUTC).map(_.toInstant).map(Date.from).foreach(builder.setDate)
+
+    validateHtmlBody.flatMap(maybeHtmlBody => {
+      if (attachments.isDefined) {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:118]: Use Option.map(...).getOrElse(...) ?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 6]
```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+      relatedMultipartBuilder.addBodyPart(bodypartBuilder.setBody(maybeHtmlBody.getOrElse(""), SubType.HTML_SUBTYPE, StandardCharsets.UTF_8))
+
+      inlineAttachments
+        .flatMap(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+        .foreach(attachmentMetadata => relatedMultipartBuilder.addBodyPart(
+          bodypartBuilder.setBody(attachmentContentLoader.load(attachmentMetadata, mailboxSession).readAllBytes(), attachmentMetadata.getType.asString())
+            .setContentDisposition("inline")))
+
+      multipartBuilder
+        .addBodyPart(BodyPartBuilder.create.setBody(maybeHtmlBody.getOrElse(""), SubType.PLAIN_SUBTYPE, StandardCharsets.UTF_8))
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:151]: Can you explain me why the html body is set up twice, one time on relatedMultipartBuilder, one time on multipartBuilder?

(we will need tests mixing attachments & textHtml BTW)

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 7]
```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+    val maybeAttachments: Either[ErrorCreatingAttachmentException, List[(AttachmentMetadata, Array[Byte])]] = otherAttachments
+      .flatMap(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+      .map(attachmentMetadata => loadAttachment(attachmentMetadata, attachmentContentLoader, mailboxSession))
+      .sequence
+
+    maybeAttachments.map(list => {
+      list.foldLeft(multipartBuilder) {
+        case (builder, (attachment, content)) =>
+          builder.addBodyPart(bodypartBuilder.setBody(content, attachment.getType.asString())
+            .setContentDisposition("attachment"))
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:164]: This needs to be attachment.disposition :-)

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 8]
```diff
@@ -76,25 +98,89 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+          builder
+      }
     })
+  }
+
+  private def getAttachmentMetadata(attachment: Attachment, attachmentManager: AttachmentManager, mailboxSession: MailboxSession): Option[AttachmentMetadata] = {
+    Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
+      .fold(_ => None, attachment => Some(attachment))
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:172]: toOption does this.

BTW using an either will prevent to "loose" the exception...

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 9]
```diff
@@ -134,52 +132,46 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+                             attachmentContentLoader: AttachmentContentLoader,
+                             mailboxSession: MailboxSession): Either[ErrorCreatingAttachmentException, (Attachment, Array[Byte])] = {
     Try(attachmentContentLoader.load(attachmentMetadata, mailboxSession))
       .toEither
-      .fold(e => {
-        //TODO log error
-        Left(ErrorCreatingAttachmentException(e.getMessage))
-      },
-        inputStream => scala.Right((attachmentMetadata, inputStream.readAllBytes())))
+      .fold(e => Left(ErrorCreatingAttachmentException(e.getMessage)),
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:173]: The stacktrace is still swollen

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 10]
```diff
@@ -1470,6 +1471,100 @@ trait EmailSetMethodContract {
...
+         |      },
+         |    "c2"]
+         |  ]
+         |}""".stripMargin
+
+    val response = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post.prettyPeek()
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:1536]: debug spotted

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 11]
```diff
@@ -61,6 +71,18 @@ case class ClientEmailBodyValue(value: String,
...
                                 isTruncated: Option[IsTruncated])
 
+case class Attachment(blobId: BlobId,
+                      `type`: Type,
+                      name: Option[Name],
+                      charset: Option[Charset],
+                      disposition: Option[Disposition],
+                      language: Option[Language],
+                      location: Option[Location],
+                      cid: Option[String]) {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:81]: Strong typing?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 12]
```diff
@@ -1555,14 +1568,29 @@ trait EmailSetMethodContract {
...
            |      "disposition": "attachment"
            |    }
            |  ]
            |}]""".stripMargin)
+
+    val downloadResponse = `given`
+      .basePath("")
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+    .when
+      .get(s"/download/$accountId/$blobIdToDownload").prettyPeek()
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:1584]: debug spotted

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 13]
```diff
@@ -134,52 +132,46 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
     })
   }
 
-  private def getAttachmentMetadata(attachment: Attachment, attachmentManager: AttachmentManager, mailboxSession: MailboxSession): Option[AttachmentMetadata] = {
+  private def getAttachmentMetadata(attachment: Attachment,
+                                    attachmentManager: AttachmentManager,
+                                    mailboxSession: MailboxSession): Either[ErrorCreatingAttachmentException, (Attachment, AttachmentMetadata)] = {
     Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
-      .fold(_ => None, attachment => Some(attachment))
+      .fold(e => Left(ErrorCreatingAttachmentException(e.getMessage)), attachmentMetadata => Right((attachment, attachmentMetadata)))
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:164]: maybe skip a line between exception and success, for visibility

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 14]
```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+  private def parametersWithoutName(typeAsField: ContentTypeField): Map[String, String] = {
+    typeAsField.getParameters
+      .asScala
+      .filter(!_._1.equals("name"))
+      .toMap
+  }
+
+  private def getAttachmentMetadata(attachment: Attachment,
+                                    attachmentManager: AttachmentManager,
+                                    mailboxSession: MailboxSession): Either[AttachmentNotFoundException, (Attachment, AttachmentMetadata)] = {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:193]: Uneeded {}

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 15]
```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+    Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
+      .fold(e => Left(AttachmentNotFoundException(e, s"Attachment not found: ${attachment.blobId.value}")),
+        attachmentMetadata => Right((attachment, attachmentMetadata)))
+  }
+
+  private def loadAttachment(attachment: Attachment,
+                             attachmentMetadata: AttachmentMetadata,
+                             attachmentContentLoader: AttachmentContentLoader,
+                             mailboxSession: MailboxSession): Either[IllegalArgumentException, (Attachment, AttachmentMetadata, Array[Byte])] = {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:202]: Uneeded {}

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 16]
```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+    Try(attachmentContentLoader.load(attachmentMetadata, mailboxSession))
+      .toEither
+      .fold(e => Left(AttachmentNotFoundException(e, s"Attachment not found: ${attachment.blobId.value}")),
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:205]: Here, attachmentContentLoader.load might throw `` & ``

You are considering IOException as not found.

Maybe a "catch" is more straightforward and less error prone?

```
try {
     val inputStream = attachmentContentLoader.load(attachmentMetadata, mailboxSession)
     scala.Right((attachment, attachmentMetadata, inputStream.readAllBytes()))
 } catch {
    case e: AttachmentNotFoundException => Left(AttachmentNotFoundException(...))
    case e: IOException => ???
}
```

And the upstream method likely need to handle IOException too... (they should fail)

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 17]
```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+      .asScala
+      .filter(!_._1.equals("name"))
+      .toMap
+  }
+
+  private def getAttachmentMetadata(attachment: Attachment,
+                                    attachmentManager: AttachmentManager,
+                                    mailboxSession: MailboxSession): Either[AttachmentNotFoundException, (Attachment, AttachmentMetadata)] = {
+    Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
+      .fold(e => Left(AttachmentNotFoundException(e, s"Attachment not found: ${attachment.blobId.value}")),
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:195]: You are considering MailboxException as "the attachment is not here".

I would suggest to switch the method type to `Either[Exception, (Attachment, AttachmentMetadata)]`

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 18]
```diff
@@ -207,6 +310,7 @@ case class ValidatedEmailSetUpdate private (keywordsTransformation: Function[Key
 class EmailUpdateValidationException() extends IllegalArgumentException
 case class InvalidEmailPropertyException(property: String, cause: String) extends EmailUpdateValidationException
 case class InvalidEmailUpdateException(property: String, cause: String) extends EmailUpdateValidationException
+case class AttachmentNotFoundException(throwable: Throwable, cause: String) extends EmailUpdateValidationException
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:313]: There is already a AttachmentNotFoundException in mailbox-api. I find this new one confusing...

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 19]
```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
...
+                                             attachmentContentLoader: AttachmentContentLoader,
+                                             mailboxSession: MailboxSession): Either[IllegalArgumentException, MultipartBuilder] = {
+    val multipartBuilder = MultipartBuilder.create(SubType.MIXED_SUBTYPE)
+    val bodypartBuilder = BodyPartBuilder.create()
+
+    attachments
+      .map(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+      .map(attachmentMetadataList => attachmentMetadataList
+        .flatMap(attachmentAndMetadata => loadAttachment(attachmentAndMetadata._1, attachmentAndMetadata._2, attachmentContentLoader, mailboxSession)))
+      .sequence
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:144]: Is : 

```
    attachments
      .map(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
      .map(attachmentMetadataList => attachmentMetadataList
        .flatMap(attachmentAndMetadata => loadAttachment(attachmentAndMetadata._1, attachmentAndMetadata._2, attachmentContentLoader, mailboxSession)))
      .sequence
```

dead code that we should remove?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 20]
```diff
@@ -26,15 +26,17 @@ import java.util.Date
 
 import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:27]: Can I have a test (Email/set + Email/get) with `attachments` & `htmlBody`?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailBodyPart.scala [hunk 21]
```diff
@@ -126,15 +126,15 @@ object EmailBodyPart {
...
+          disposition = Option(entity.getDispositionType).map(Disposition(_)),
           cid = headerValue(entity, "Content-Id")
             .flatMap(Cid.parser()
               .relaxed()
               .unwrap()
               .parse(_)
               .toScala),
           language = headerValue(entity, "Content-Language")
-            .map(Language),
+            .map(_.split("; ").toList.map(Language)),
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailBodyPart.scala:137]: This should be isolated in a method of Language

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 22]
```diff
@@ -1720,6 +1723,511 @@ trait EmailSetMethodContract {
...
+           |      "size": 11,
+           |      "type": "text/plain",
+           |      "charset": "UTF-8",
+           |      "disposition": "attachment"
+           |    }
+           |  ],
+           |  "htmlBody": [
+           |    {
+           |      "partId": "2",
+           |      "blobId": "1_2",
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:1967]: Should be `"blobId": "${messageId}_2"` or should be ommited in `whenIgnoringPaths`...

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailGetSerializer.scala [hunk 23]
```diff
@@ -21,7 +21,7 @@ package org.apache.james.jmap.json
 
 import org.apache.james.jmap.api.model.Preview
 import org.apache.james.jmap.core.Properties
-import org.apache.james.jmap.mail.{AddressesHeaderValue, BlobId, Charset, DateHeaderValue, Disposition, EmailAddress, EmailAddressGroup, EmailBody, EmailBodyMetadata, EmailBodyPart, EmailBodyValue, EmailFastView, EmailFullView, EmailGetRequest, EmailGetResponse, EmailHeader, EmailHeaderName, EmailHeaderValue, EmailHeaderView, EmailHeaders, EmailIds, EmailMetadata, EmailMetadataView, EmailNotFound, EmailView, EmailerName, FetchAllBodyValues, FetchHTMLBodyValues, FetchTextBodyValues, GroupName, GroupedAddressesHeaderValue, HasAttachment, HeaderMessageId, HeaderURL, IsEncodingProblem, IsTruncated, Keyword, Keywords, Language, Location, MailboxIds, MessageIdsHeaderValue, Name, PartId, RawHeaderValue, Subject, TextHeaderValue, ThreadId, Type, URLsHeaderValue}
+import org.apache.james.jmap.mail._
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailGetSerializer.scala]: oups

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 24]
```diff
@@ -1720,6 +1723,494 @@ trait EmailSetMethodContract {
...
+    val bobPath = MailboxPath.inbox(BOB)
+    val mailboxId = server.getProbe(classOf[MailboxProbeImpl]).createMailbox(bobPath)
+    val payload = "123456789\r\n".getBytes(StandardCharsets.UTF_8)
+
+    val uploadResponse: String = `given`
+      .basePath("")
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .contentType("text/plain")
+      .body(payload)
+      .when
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:1992]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 25]
```diff
@@ -1720,6 +1723,494 @@ trait EmailSetMethodContract {
...
+      .post(s"/upload/$ACCOUNT_ID/")
+      .`then`
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:1994]: indent

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**